packages/atlas/src/api/hooks/nfts.ts

```diff
@@ -1,10 +1,82 @@
-import { useGetNftQuery } from '@/api/queries'
+import {
+  VideoCategoryWhereInput,
+  VideoOrderByInput,
+  useGetNftQuery,
+  useGetNftsQuery,
+  useGetVideosConnectionQuery,
+} from '@/api/queries'
+import { createLookup } from '@/utils/data'
 
-export const useNft = (videoId: string) => {
-  const { data, ...rest } = useGetNftQuery({ variables: { videoId: videoId }, skip: !videoId })
+export const useNft = (id: string) => {
+  const { data, ...rest } = useGetNftQuery({ variables: { id }, skip: !id })
 
   return {
-    nft: data?.ownedNfts[0],
+    nft: data?.ownedNftByUniqueInput,
+    ...rest,
+  }
+}
+
+type ChannelNftsOpts = {
+  category?: VideoCategoryWhereInput | null
+  sortNftsBy?: VideoOrderByInput
+}
+
+export const useChannelNfts = (channelId: string, opts?: ChannelNftsOpts) => {
+  const {
+    data: videosConnectionData,
+    loading: videosConnectionLoading,
+    ...rest
+  } = useGetVideosConnectionQuery({
+    variables: {
+      orderBy: opts?.sortNftsBy,
+      where: {
+        channel: { id_eq: channelId },
+        nft: {
+          metadata_contains: '',
+        },
+        isPublic_eq: true,
+        isCensored_eq: false,
+
+        thumbnailPhoto: {
+          isAccepted_eq: true,
+        },
+        media: {
+          isAccepted_eq: true,
+        },
+        ...(opts?.category?.id_in?.length
+          ? {
+              category: {
+                id_in: opts.category.id_in,
+              },
+            }
+          : {}),
+      },
+    },
+  })
+
+  const idIn = videosConnectionData?.videosConnection.edges.map((edge) => edge.node.id) || []
+
+  const { data: nftsData, loading: nftsLoading } = useGetNftsQuery({
+    variables: {
+      where: {
+        id_in: idIn,
+      },
+    },
+    skip: !idIn.length || videosConnectionLoading,
+  })
```

Comment:
This is temporary, yes? I'd assume once we can filter NFTs by owner, we will get rid of the videos query? If that's the case, could you add `TODO` comment?

Reply:
yes, temporary - I'll add a comment